### PR TITLE
refactor(Apps, Deps): Remove Deno as a dependency to the dashboard

### PR DESCRIPTION
**Pull request by michaeldelago**
## Changes Proposed:
- This PR removes Deno as a dependency.
  - Roughly 400 lines of TypeScript have been removed

- `apps/ci/ci-pending-changelogs.ts` was previously wrapped by a bash script and called during github actions. This has been replace with the github action install `deno` and just running `deno` on the proper file path

- `apps/docker/docker-cmd.ts` has been rewritten to a shell script with similar functionality
- `apps/bash_shared/deno.sh` has been removed
- `apps/bash_shared/includes.sh` has had deno related steps removed
- `apps/installer/main.ts` has been removed
- `apps/installer/utils.ts` has been removed
  - this file and the one above were only used for reading `acore.json` and printing the version of AzerothCore.
- `deps/deno` has been removed

## Issues Addressed:

- in PR https://github.com/azerothcore/azerothcore-wotlk/pull/16898, work was put into removing Deno from the docker setup. 
- through working through that PR, I felt stronger and stronger that the addition of deno is unnecessary
- resolves issues stemming from the use of Deno
    - https://github.com/azerothcore/azerothcore-wotlk/issues/6131
    - https://github.com/azerothcore/azerothcore-wotlk/issues/9071

## Tests Performed:

- The only high level changes here are for `./acore.sh docker ...` and `./acore.sh version`
- `./acore.sh docker build client-data start:app` works
- `./acore.sh version` works

## How to Test the Changes:

1. Fairly straightforward. Execute these commands: 
  a. `./acore.sh docker build client-data start:app`
  b. `./acore.sh version`

## Known Issues and TODO List:
- Technically speaking, behavior for the `./acore.sh docker ...` changes
  - I made it just `exit 1` if a command is not provided (And showing all the available options)
  - I doubt that this will be an issue, but I can the implement captive shell into it if needed.
- This can replace https://github.com/azerothcore/azerothcore-wotlk/pull/16898 or be merged after it

<details>

<summary>rambling about the vision I have for the docker setup</summary> 

- From a higher level, it'd probably be better for us to "unwrap" the docker builds from `./acore.sh`
  - Docker-based infra for an application should be simple and familiar
  - The `./acore.sh` dashboard is unique only to azerothcore
  - Both using `./acore.sh` and working on it require knowledge of lots of "external" information. ie, the thought "oh thats in this file, and var A is not in this file but in that file"
  - `./acore.sh` is a soft requirement for running azerothcore through docker
    - there's a 2 of each service in the compose file (and in the Dockerfile) with a `dev` and `prod` distinction
    - This distinction isn't made clear to the person trying to build the containers
    - The `dev` containers are the "default" containers that are documented The binaries are built with the `ac-dev-server` service, and the compiled binaries are placed into a volume that is then mounted by the runtime containers. This leads to a confusing/non-standard build process.
    - the `prod` containers are almost entirely identical, except that [eluna](https://github.com/azerothcore/mod-eluna) is installed by default and the binaries are built as a part of the Dockerfile.
    - These containers are those used for [acore-docker](https://github.com/azerothcore/acore-docker), which is a couple pre-built containers that act as a "repack" or quick-start for users
  - In many projects, seeing a compose file allows the safe assumption that a user can simply clone and run `docker compose up -d --build`
  - This assumption doesn't ring true for azerothcore and I think that should be fixed

</details>

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by michaeldelago** [comment]
@Kitzunu how do you feel about closing #16898 and going off of this one instead?

**Comment by Kitzunu** [comment]
sure

**Comment by michaeldelago** [comment]
This has been successfully tested on an AWS EC2 instance (type c6g) running Amazon Linux 3 (roughly equivalent to Fedora). The below commands procedure as expected:

1. `./acore.sh docker build`
2. `./acore.sh docker client-data`
3. ./acore.sh docker start:app`

**Comment by ben-of-codecraft** [comment]
This PR is working find to build and run fixes issues a few of us are having trying to use the docker build, I also. love it kills of the abstraction on the existing sh scripts. 

Built on ubuntu 20

File: .github/workflows/import_pending.yml [hunk 1]
```diff
@@ -14,21 +14,29 @@ jobs:
...
+          # Specifies latest 1.x
+          deno-version: "~1.0"
+
       - name: Import and commit pending sql
         run: |
           git config user.email "azerothcorebot@gmail.com" && git config user.name "AzerothCoreBot"
           bash bin/acore-db-pendings
-          bash bin/acore-import-changelogs
-          git fetch --unshallow origin ${BRANCH}
+          deno apps/ci/ci-pending-changelogs.ts
```
Comment (Kitzunu) [inline .github/workflows/import_pending.yml:27]: Just help me understand why we are changing pending changelog and not pending db?

Comment (michaeldelago) [reply .github/workflows/import_pending.yml:31]: `bin/acore-db-pendings` is a shell script, and doesnt have any typescript to be used with deno. After it starts, it sources another shell script (`apps/ci/ci-pending-sql.sh`) and doesn't use any typescript

`bin/acore-import-changelogs` sources `apps/bash_shared/includes.sh`. This installs Deno. It then executes the typescript file at `apps/ci/ci-pending-changelogs.ts`. Proxying the typescript file through the `acore-import-changelogs` shell script doesn't seem to add anything and it makes the execution a bit harder to trace. having the CI directly run `deno apps/ci/ci-pending-changelogs.ts` removes an unnecessary abstraction.

Comment (Kitzunu) [reply .github/workflows/import_pending.yml:31]: kk

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**